### PR TITLE
Add support for the user.getTopTracks endpoint

### PR DESCRIPTION
**Pull request by alch-emi**
This pull request adds support for the user.getTopTracks endpoint.

Aside from the new `user::top_tracks` endpoint, this adds an optional `playcount` attribute to track information, which is part of the returned values by the getTopTracks endpoint.

A change that might need to be made is that the `Period` enum currently in the `user::top_artists` module is also used for this endpoint.  It may make sense to move this enum into the model module, but this change was not included in this pull request in order to keep things simple.

Please let me know if you have any feedback or suggestions.

### CODE REVIEW

**Comment by alch-emi** [comment]
A quick addendum, this doesn't include a test for the new method because I got a little overwhelmed looking at the other test data.  If you have any information on the best way to go about synthesizing test data, I will be happy to add tests

**Comment by drklee3** [comment]
You can make a test json file for top tracks by downloading the raw json response data from the top tracks endpoint, eg `curl https://ws.audioscrobbler.com/2.0/?method=user.getTopTracks&user=...&api_key=...&format=json -o user.getTopTracks.json`

**Comment by alch-emi** [comment]
oh yeah i suppose that makes sense doesn't it :p

thank you for the tip!  i just added the tests now

File: tests/deserialize.rs [hunk 1]
```diff
@@ -34,3 +34,8 @@ fn user_get_info() {
 fn user_get_loved_tracks() {
     p!(User, "user.getLovedTracks");
 }
+
+#[test]
+fn user_get_top_tracks() {
+    p!(User, "user.getLovedTracks");
```
Comment (drklee3) [inline tests/deserialize.rs]:
```suggestion
    p!(User, "user.getTopTracks");
```

**Comment by evieluvsrainbows** [comment]
Thank you!

**Pull request merged**